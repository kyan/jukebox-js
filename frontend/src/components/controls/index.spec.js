import React from 'react'
import { shallow } from 'enzyme'
import MopidyApi from '../../constants/mopidy-api'
import Controls from './index'

describe('Controls', () => {
  const onPlayMock = jest.fn()
  const onStopMock = jest.fn()
  const onPauseMock = jest.fn()
  const onPrevMock = jest.fn()
  const onNextMock = jest.fn()

  describe('render', () => {
    beforeEach(() => {
      jest.clearAllMocks()
    })

    describe('when paused', () => {
      const wrapper = shallow(
        <Controls
          playbackState={MopidyApi.PAUSED}
          disabled={false}
          onPlay={onPlayMock}
          onStop={onStopMock}
          onPause={onPauseMock}
          onPrevious={onPrevMock}
          onNext={onNextMock}
        />
      )

      it('renders as expected', () => {
        expect(wrapper).toMatchSnapshot()
      })
    })

    describe('when stopped', () => {
      const wrapper = shallow(
        <Controls
          playbackState={MopidyApi.STOPPED}
          disabled={false}
          onPlay={onPlayMock}
          onStop={onStopMock}
          onPause={onPauseMock}
          onPrevious={onPrevMock}
          onNext={onNextMock}
        />
      )

      it('renders as expected', () => {
        expect(wrapper).toMatchSnapshot()
      })
    })

    describe('when disabled', () => {
      const wrapper = shallow(
        <Controls
          playbackState={MopidyApi.PLAYING}
          disabled
          onPlay={onPlayMock}
          onStop={onStopMock}
          onPause={onPauseMock}
          onPrevious={onPrevMock}
          onNext={onNextMock}
        />
      )

      it('renders as expected', () => {
        expect(wrapper).toMatchSnapshot()
      })
    })

    describe('when no playstate', () => {
      const wrapper = shallow(
        <Controls
          disabled={false}
          onPlay={onPlayMock}
          onStop={onStopMock}
          onPause={onPauseMock}
          onPrevious={onPrevMock}
          onNext={onNextMock}
        />
      )

      it('renders as expected', () => {
        expect(wrapper).toMatchSnapshot()
      })
    })

    describe('when playing', () => {
      const wrapper = shallow(
        <Controls
          playbackState={MopidyApi.PLAYING}
          disabled={false}
          onPlay={onPlayMock}
          onStop={onStopMock}
          onPause={onPauseMock}
          onPrevious={onPrevMock}
          onNext={onNextMock}
        />
      )

      it('renders as expected', () => {
        expect(wrapper).toMatchSnapshot()
      })

      it('handles a play click', () => {
        wrapper.find('.jb-play-button').simulate('click')
        expect(onPlayMock).toHaveBeenCalled()
      })

      it('handles a stop click', () => {
        wrapper.find('.jb-stop-button').simulate('click')
        expect(onStopMock).toHaveBeenCalled()
      })

      it('handles a pause click', () => {
        wrapper.find('.jb-pause-button').simulate('click')
        expect(onPauseMock).toHaveBeenCalled()
      })
    })
  })
})
