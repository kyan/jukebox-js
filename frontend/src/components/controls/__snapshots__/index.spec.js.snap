// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls render when controls disabled renders as expected 1`] = `
<span>
  <Button
    active={false}
    as="button"
    className="jb-radio-button"
    color={null}
    disabled={true}
  >
    <ButtonContent
      hidden={true}
    >
      RADIO 
    </ButtonContent>
  </Button>
  <SkipButtons
    disabled={true}
    onNext={[Function]}
    onPrevious={[Function]}
  />
  <Button
    active={false}
    animated="vertical"
    as="button"
    className="jb-play-button"
    disabled={true}
    onClick={[Function]}
  >
    <ButtonContent
      hidden={true}
    >
      Play
    </ButtonContent>
    <ButtonContent
      visible={true}
    >
      <Icon
        as="i"
        name="play"
      />
    </ButtonContent>
  </Button>
  <Button
    active={false}
    animated="vertical"
    as="button"
    className="jb-pause-button"
    disabled={true}
    onClick={[Function]}
  >
    <ButtonContent
      hidden={true}
    >
      Pause
    </ButtonContent>
    <ButtonContent
      visible={true}
    >
      <Icon
        as="i"
        name="pause"
      />
    </ButtonContent>
  </Button>
</span>
`;

exports[`Controls render when controls enabled renders as expected 1`] = `
<span>
  <Button
    active={false}
    as="button"
    className="jb-radio-button"
    color={null}
    disabled={false}
  >
    <ButtonContent
      hidden={true}
    >
      RADIO 
    </ButtonContent>
  </Button>
  <SkipButtons
    disabled={false}
    onNext={[Function]}
    onPrevious={[Function]}
  />
  <Button
    active={false}
    animated="vertical"
    as="button"
    className="jb-play-button"
    disabled={false}
    onClick={[Function]}
  >
    <ButtonContent
      hidden={true}
    >
      Play
    </ButtonContent>
    <ButtonContent
      visible={true}
    >
      <Icon
        as="i"
        name="play"
      />
    </ButtonContent>
  </Button>
  <Button
    active={false}
    animated="vertical"
    as="button"
    className="jb-pause-button"
    disabled={false}
    onClick={[Function]}
  >
    <ButtonContent
      hidden={true}
    >
      Pause
    </ButtonContent>
    <ButtonContent
      visible={true}
    >
      <Icon
        as="i"
        name="pause"
      />
    </ButtonContent>
  </Button>
</span>
`;

exports[`Controls render when streaming radio renders as expected 1`] = `
<span>
  <Button
    active={true}
    as="button"
    className="jb-radio-button"
    color="green"
    disabled={false}
  >
    <ButtonContent
      hidden={true}
    >
      RADIO 
      <Icon
        as="i"
        loading={true}
        name="certificate"
      />
    </ButtonContent>
  </Button>
  <SkipButtons
    disabled={false}
    onNext={[Function]}
    onPrevious={[Function]}
  />
  <Button
    active={false}
    animated="vertical"
    as="button"
    className="jb-play-button"
    disabled={false}
    onClick={[Function]}
  >
    <ButtonContent
      hidden={true}
    >
      Play
    </ButtonContent>
    <ButtonContent
      visible={true}
    >
      <Icon
        as="i"
        name="play"
      />
    </ButtonContent>
  </Button>
  <Button
    active={false}
    animated="vertical"
    as="button"
    className="jb-pause-button"
    disabled={false}
    onClick={[Function]}
  >
    <ButtonContent
      hidden={true}
    >
      Pause
    </ButtonContent>
    <ButtonContent
      visible={true}
    >
      <Icon
        as="i"
        name="pause"
      />
    </ButtonContent>
  </Button>
</span>
`;
